### DISCUSSION TITLE
getting neighbour list from a list of bonds in freud

### DISCUSSION THREAD

**Discussion by mubisk** [glotzerlab/freud#1200] [Support] [answered]
Hi. I am new to Freud and would like to calculate cluster properties. I have ring polymeric system and would like to form cluster only based on the bonds. However I a stuck with the following command:
cl = freud.cluster.Cluster()
cl.compute(system, neighbors)

I tried converting my array of bonds into a dictionary and using as neighbors in the above command but I am getting the following error: "ValueError: An invalid value was provided for neighbors, which must be a dict or NeighborList object."

The bonds array in my code is [[1,2], [2,3], [3,4], .......].

Can someone please help me in formatting the neighbors correctly?

**Comment by tommy-waltmann** [comment]
Hi @mubisk,

For the purpose of organization, please ask this question on the [freud discussion board](https://github.com/glotzerlab/freud/discussions).

**Reply by joaander** [reply]
I transferred the discussion from hoomd-blue to freud.

**Comment by tommy-waltmann** [comment, marked as answer]
To create a neighbor list for which the neighbors are already known, use the factory method `from_arrays` of the `NeighborList` class. There is an example and further description in the documentation [here](https://freud.readthedocs.io/en/latest/modules/locality.html#freud.locality.NeighborList.from_arrays)

If you have any further questions, don't hesistate to ask!
